### PR TITLE
Fe#45 error handling api implementation

### PR DESCRIPTION
**Pull request by stripes416**

### CODE REVIEW

File: js/index.js [hunk 1]
```diff
@@ -31,23 +31,35 @@ $(function() {
...
+            "url": backendIpAddress + "login",
+            "data": JSON.stringify({
                 "username": $('#loginUsername').val(),
-                "password": $('#loginPassword').val(),
-            },
-            type: "GET"
+                "password": $('#loginPassword').val()
+            }),
+            "type": "POST"
         }).done(function(json) {
```
Comment (SoySauceLabs) [inline js/index.js:40]: add a comma to the end of line 39

File: js/index.js [hunk 2]
```diff
@@ -31,23 +31,35 @@ $(function() {
...
-                $("#loginError").text("Error logging in.");
+              var responseObject = JSON.parse(json.responseText);
+              if (responseObject && responseObject.errors) {
+                var errors = responseObject.errors;
+                for (var errorCode in errors) {
+                $("#loginError").text(errors[errorCode]);
+                }
+              } else {
+                $("loginError").text("An unknown error occured. Please try again.");
             }
```
Comment (SoySauceLabs) [inline js/index.js:53]: Add a new line after line 53 and add a **}** to it

File: js/index.js [hunk 3]
```diff
@@ -31,23 +31,35 @@ $(function() {
...
+        }).fail(function(json) {
+            var responseObject = JSON.parse(json.responseText);
+            if (responseObject && responseObject.errors) {
+              var errors = responseObject.errors;
+              for (var errorCode in errors) {
+                $("#loginError").text(errors[errorCode]);
+              }
+            } else {
+                $("loginError").text("An unknown error occured. Please try again.");
         });
```
Comment (SoySauceLabs) [inline js/index.js:63]: Add a new line after line 62 and add a **}** to it

**Comment by SoySauceLabs** [comment]
Just a couple small items to address. Otherwise it looks good. Although this PR needs to be re-created with **develop** as the base branch. 👍

**Comment by stripes416** [comment]
https://github.com/Plantalytics/plantalytics-frontend/pull/63

**Pull request closed without merging**